pkg/controller/machinedeployment/controller_test.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package machinedeployment
+
+import (
+	"reflect"
+	"testing"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+func TestMachineSetToDeployments(t *testing.T) {
+	machineDeplopymentList := &v1alpha1.MachineDeploymentList{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "MachineDeploymentList",
+		},
+		Items: []v1alpha1.MachineDeployment{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "withMatchingLabels",
+					Namespace: "test",
+				},
+				Spec: v1alpha1.MachineDeploymentSpec{
+					Selector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"foo": "bar",
+						},
+					},
+				},
+			},
+		},
+	}
+	controller := true
+	ms := v1alpha1.MachineSet{
```

Comment:
Given that we enumerate the other instances (ms2, ms3), why not make this `ms1`?